### DISCUSSION TITLE
Website not starting with the following error Cannot find module 'commander'

### DISCUSSION THREAD

**Discussion by jxbuck** [rdudhagra/Sand-Table#14] [Troubleshooting] [answered]
I'm seeing this when attempting to run prod for the first time after install.  How might I go about resolving this? commander seems to be installed ok. Is it a version issue perhaps?

Error: Cannot find module 'commander'
Require stack:
- /home/pi/Website/node_modules/@babel/cli/lib/babel/options.js
- /home/pi/Website/node_modules/@babel/cli/lib/babel/index.js
- /home/pi/Website/node_modules/@babel/cli/bin/babel.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1080:15)
    at Function.Module._load (internal/modules/cjs/loader.js:923:27)
    at Module.require (internal/modules/cjs/loader.js:1140:19)
    at require (internal/modules/cjs/helpers.js:75:18)
    at _commander (/home/pi/Website/node_modules/@babel/cli/lib/babel/options.js:19:39)
    at Object.<anonymous> (/home/pi/Website/node_modules/@babel/cli/lib/babel/options.js:52:1)
    at Module._compile (internal/modules/cjs/loader.js:1251:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1272:10)
    at Module.load (internal/modules/cjs/loader.js:1100:32)
    at Function.Module._load (internal/modules/cjs/loader.js:962:14) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/pi/Website/node_modules/@babel/cli/lib/babel/options.js',
    '/home/pi/Website/node_modules/@babel/cli/lib/babel/index.js',
    '/home/pi/Website/node_modules/@babel/cli/bin/babel.js'
  ]

![image](https://user-images.githubusercontent.com/82311012/114297981-fb269300-9a81-11eb-9a71-fb0519a56340.png)

**Comment by jxbuck** [comment, marked as answer]
I think the bigger issue I'm running into is the age of the build compared to the current version of a lot of the packages.  I'm relatively new to the pi and my Debian knowledge is 20 years out of date.  But I'm updating that knowledge.  I think my question is- how is this typically handled?  Can anyone point me to a resource?  Am i correct to simply keep chasing each problematic package and installing it manually then reattempting a build?.  I humbly thank you in advance

**Reply by rdudhagra** [reply]
Huh, yet another weird `npm` error I haven't seen before. This is definitely more of a NodeJS issue and not a Debian issue. Try running `npm install commander --save` and then `npm install` again...see if that works ([source](https://stackoverflow.com/questions/49393301/error-cannot-find-module-commander))

"Am i correct to simply keep chasing each problematic package and installing it manually then reattempting a build?"
-- as far as I know, yes unfortunately

Let me know if that works for you

**Reply by rdudhagra** [reply]
Oh didn't see above where you checked that `commander` is installed correctly: I just checked my Pi and the npm version seems to be different...
Try running `npm install -g npm@6.14.8` and see if that fixes the issue. NodeJS and npm are notorious for breaking with version upgrades, so a lot teams that use this framework basically just develop with a fixed version number and only upgrade if necessary. If this ends up solving the issue, let me know and I'll make a note of this in the build guide.

**Reply by jxbuck** [reply]
Great.  After an update to babel (npm i @babel/cli@7.8.3), that did it!   I'm seeing the website. Thanks.  

I did notice that I got an error on the first track uploaded and it killed the server.  The second track worked just fine.  I'll dig into this error at another time.   I think I might be resolved when I make the packet size changes to the node file.

Thanks!

**Reply by rdudhagra** [reply]
No problem! Let me know if the issue with the file uploading still exists...could be that the first file has some comment/tag that I am not parsing out in my code
